io/plugins/it.geosolutions.hale.io.appschema.test/src/it/geosolutions/hale/io/appschema/writer/AppSchemaMappingTest.java

```diff
@@ -320,6 +321,15 @@ public void testNestedJoinHandler() {
 		assertEquals(SOURCE_DATASET_ID, nestedMapping.getSourceExpression().getOCQL());
 		assertNull(nestedMapping.getSourceExpression().getLinkElement());
 		assertNull(nestedMapping.getSourceExpression().getLinkField());
+
+		// check namespace correct prefix output
+		List<Namespace> namespaces = mainMapping.getNamespaces().getNamespace();
+		Namespace baseNamespace = namespaces.stream()
+				.filter(namespace -> "http://inspire.ec.europa.eu/schemas/base/3.3"
+						.equals(namespace.getUri()))
+				.findFirst().orElse(null);
+		assertNotNull(baseNamespace);
+		assertEquals("base", baseNamespace.getPrefix());
```

Comment:
You check here should be more assertive, and IMHO you should create an helper method for that that gets a map of namespaces and prefix and check that they are present on the mapping. You should have a boolean to control if the check should be exact or not in terms of number of namespaces.